webknossos/webknossos/dataset/dataset.py

```diff
@@ -967,18 +980,30 @@ def add_layer_for_existing_files(
         assert (
             array_info is not None
         ), f"Could not find any valid mags in {self.path /layer_name}. Cannot add layer."
+
+        num_channels = kwargs.pop("num_channels", array_info.num_channels)
+        dtype_per_channel = kwargs.pop("dtype_per_channel", array_info.voxel_type)
+        data_format = kwargs.pop("data_format", array_info.data_format)
+
         layer = self.add_layer(
             layer_name,
             category=category,
-            num_channels=array_info.num_channels,
-            dtype_per_channel=array_info.voxel_type,
-            data_format=array_info.data_format,
+            num_channels=num_channels,
+            dtype_per_channel=dtype_per_channel,
+            data_format=data_format,
             **kwargs,
         )
         for mag_dir in layer.path.iterdir():
+            try:
+                Mag(mag_dir.name)  # test if folder name is a valid mag
+            except ValueError:
+                continue
```

Comment:
could you add a comment explaining which expected cases should trigger this continue?